db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/CastNode.java

```diff
@@ -300,6 +300,8 @@ public ValueNode bindExpression(FromList fromList,
             castOperand.setType(getTypeServices());
         }
 
+        addSPSPropertyDependency();
```

Comment:
1. I'm not sure if at this point we know the target type of the cast. We bind only the operand and not yet the cast node itself.
2. bindExpression() is called only for explicit casts users write in queries. If a `CastNode` is added in our binding phase as an implicit cast, we will just call `((CastNode) newNode).bindCastNodeOnly()`. If such a cast happen to trigger a dependency, we will miss it. Question is if that should be how it works by design.

Reply:
> 1. I'm not sure if at this point we know the target type of the cast. We bind only the operand and not yet the cast node itself.
> 2. bindExpression() is called only for explicit casts users write in queries. If a CastNode is added in our binding phase as an implicit cast, we will just call ((CastNode) newNode).bindCastNodeOnly(). If such a cast happen to trigger a dependency, we will miss it. Question is if that should be how it works by design.

yes such a cast could cause a dependency trigger, for example `upper(timestamp('2020-10-10 01:02:03.123456'))` will create such a cast and it is important to create a dependency to e.g. timestamp format SPS property. I moved the invocation of `addSPSPropertyDependency()` from `bindExpression` to `bindCastNodeOnly` and removed it from `com.splicemachine.db.impl.sql.compile.SimpleStringOperatorNode#bindExpression` since it is redundant.